docs/pages/guides/typescript.mdx

```diff
@@ -0,0 +1,83 @@
+import PageMeta from '_shared/PageMeta';
+
+<PageMeta
+  title="Typescript – @material-ui/pickers"
+  description="A guide of how to use @material-ui/pickers with typescript"
+/>
+
+## TypeScript guide
+
+Pickers are originally written in typescript, so type definetions are provided with the code without any additional requirements.
+Here is some information about how to achive better experience with typescript.
+
+<!-- https://stackoverflow.com/questions/62452160/material-datetimepicker-onchange-typescript-mismatch-error/62471345 -->
+
+#### Nullable value
+
+Any picker can return null in the `onChange` argument so if you are using `React.useState`, you **must** manually cast your default value as nullable. (e.g. `Date | null`)
+
+```tsx
+const MyComponent = () => {
+  const [date, setDate] = React.useState<Date | null>(new Date());
+
+  return (
+    <DatePicker
+      value={date}
+      onChange={date => setDate(date)}
```

Comment:
It might be best to avoid the implicit returned value, I have already been leaning toward it to ease refactorization, but it doesn't make a huge different

```suggestion
      onChange={date => { setDate(date) }}
```